backend/coreapp/migrations/0042_migrate_presets_to_db.py

```diff
@@ -823,6 +823,7 @@ def create_presets(apps: Apps, schema_editor: BaseDatabaseSchemaEditor) -> None:
 class Migration(migrations.Migration):
     dependencies = [
         ("coreapp", "0041_remove_project_and_github_stuff"),
+        ("coreapp", "0054_presets_owner"),
```

Comment:
is this intentional? I'm not a migration expert, but I don't think an older migration should depend on a newer one

Reply:
There is a problem with the migration, and I was not sure how to solve it...

The `0042_migrate_presets_to_db.py` is adding items to the Preset table, however they are using the latest `Preset` model class, which has the `owner` column, therefore leading to an error, when step 42 is triggered before step 54, which add the column...

I am not a migration expect either, and would appreciate some help on this particular issue!

Reply:
This behaviour seems to have already been done in previous migration:

https://github.com/decompme/decomp.me/blob/6efceba4302d7064ec3d0614c5ed0bd12aed9424/backend/coreapp/migrations/0003_github_profile.py#L11

Reply:
I believe the way to handle this is the migration code shouldn't actually be using the real model classes, but I'm not 100% sure. we probably have screwed this up in the past. I'll see if I can take a look later today

Reply:
Keep me updated